server/notification-providers/nostr.js

```diff
@@ -0,0 +1,108 @@
+const NotificationProvider = require("./notification-provider");
+global.crypto = require("crypto");
+
+const {
+    relayInit,
+    getPublicKey,
+    getEventHash,
+    signEvent,
+    nip04,
+    nip19
+} = require("nostr-tools");
+require("websocket-polyfill");
```

Comment:
Why is this included?
No other of the providers include this.

If you think this is needed, let's split this out of this PR.

Reply:
Without it, on Node 18 there is an error from the nostr-tools module. Node 19+ doesn't need it.
```
TypeError: Cannot read properties of undefined (reading 'readyState')
    at Object.close (/home/ansible/uptime-kuma/node_modules/nostr-tools/lib/nostr.cjs.js:566:14)
    at Nostr.send (/home/ansible/uptime-kuma/server/notification-providers/nostr.js:66:23)
    at async Socket.<anonymous> (/home/ansible/uptime-kuma/server/server.js:1209:27)
```

Reply:
Please include a comment that this is a `polyfill for node <=  18`